python/cudf/cudf/dataframe/numerical.py

```diff
@@ -49,16 +49,20 @@ def __contains__(self, item):
         Returns True if column contains item, else False.
         """
         # Handles improper item types
-        try:
-            item = float(item)
-        except Exception:
-            """It means that the item was not a numerical type"""
+        if np.can_cast(item, self.data.mem.dtype):
+            item = self.data.mem.dtype.type(item)
+        else:
             return False
-
-        return (
-            cudautils.find_first(self.astype("float_").data.mem, float(item))
-            != -1
-        )
+        # Issue with cudautils with bool araray, always returns True.
+        if self.data.mem.dtype == np.bool:
+            return (
+                cudautils.find_first(
+                    self.astype("int_").data.mem, np.int_(item)
+                )
+                != -1
+            )
```

Comment:
@kkraus14 And with new libcudf functionality this will not be a problem and I will be removing this segment of code.

Reply:
Understood, you should be able to view this as an int8 array instead of typecasting to an int64 array to accomplish the same thing without the expense.
```
self.data.mem.view('int8'), item.view('int8')
```

Reply:
Thank you, I wasn't aware of this option.